From 0b68f91cd05781fbc5038f1e26882b38bb1d63ef Mon Sep 17 00:00:00 2001
From: Limeng <Meng.Li@windriver.com>
Date: Tue, 7 Jan 2020 10:26:17 +0800
Subject: [PATCH 149/151] arch: arm64: dts: add dma-ranges property for pcie
 feature

It is need to limit the DMA on lower 2GB region only with
dma-ranges property in dts node. Because ARM Cortex-A53, L3
interconnect peripherals and SDM can only access the first
2 GB of this 4 GB region. In this way,PCIe feature works
fine when system has 4G memory space.

This modification refers to patch from Intel technical support.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
Integrated-by: Jun Zhang <jun.zhang@windriver.com>
---
 arch/arm64/boot/dts/altera/socfpga_stratix10_socdk_pcie.dts | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/altera/socfpga_stratix10_socdk_pcie.dts b/arch/arm64/boot/dts/altera/socfpga_stratix10_socdk_pcie.dts
index 1d84c45da68f..9b1dea42f840 100644
--- a/arch/arm64/boot/dts/altera/socfpga_stratix10_socdk_pcie.dts
+++ b/arch/arm64/boot/dts/altera/socfpga_stratix10_socdk_pcie.dts
@@ -14,8 +14,9 @@
 			reg-names = "axi_h2f", "axi_h2f_lw";
 			#address-cells = <2>;
 			#size-cells = <1>;
+			dma-ranges = <0x00000000 0x00000000 0x00000000 0x80000000>;
 			ranges = <0x00000000 0x00000000 0x80000000 0x00040000>,
-				<0x00000000 0x10000000 0x90000000 0x08000000>,
+				<0x00000000 0x10000000 0x90000000 0x10000000>,
 				<0x00000000 0x20000000 0xa0000000 0x00200000>,
 				<0x00000001 0x00010000 0xf9010000 0x00008000>,
 				<0x00000001 0x00018000 0xf9018000 0x00000080>,
@@ -37,6 +38,8 @@
 				msi-parent = <&pcie_0_msi_irq>;
 				#address-cells = <3>;
 				#size-cells = <2>;
+				dma-coherent;
+				dma-ranges = <0x02000000 0 0 0 0 0 0x80000000>;
 				interrupt-map-mask = <0 0 0 7>;
 				interrupt-map = <0 0 0 1 &pcie_0_pcie_s10 1>,
 						<0 0 0 2 &pcie_0_pcie_s10 2>,
-- 
2.26.1

